src/log_surgeon/finite_automata/RegexNFA.hpp

```diff
@@ -42,6 +91,16 @@ class RegexNFAState {
 public:
     using Tree = UnicodeIntervalTree<RegexNFAState*>;
 
+    RegexNFAState() = default;
+
+    explicit RegexNFAState(uint32_t const tag, RegexNFAState const* dest_state) {
+        m_positive_tagged_transitions.emplace_back(tag, dest_state);
+    }
+
+    explicit RegexNFAState(std::set<uint32_t> tags, RegexNFAState const* dest_state) {
+        m_negative_tagged_transitions.emplace_back(std::move(tags), dest_state);
+    }
```

Comment:
```suggestion
    RegexNFAState(uint32_t const tag, RegexNFAState const* dest_state) {
        m_positive_tagged_transitions.emplace_back(tag, dest_state);
    }

    RegexNFAState(std::set<uint32_t> tags, RegexNFAState const* dest_state) {
        m_negative_tagged_transitions.emplace_back(std::move(tags), dest_state);
    }
```
We only need to add `explicit` if the constructor only takes one input. It's to avoid implicit type conversion

Reply:
Oops, forgot to switch it when I added a second argument.